Signal/src/call/UserInterface/CallUIAdapter.swift

```diff
@@ -30,24 +30,29 @@ protocol CallUIAdaptee {
 // Shared default implementations
 extension CallUIAdaptee {
     internal func showCall(_ call: SignalCall) {
+        AssertIsOnMainThread()
+
         let callNotificationName = CallService.callServiceActiveCallNotificationName()
         NotificationCenter.default.post(name: NSNotification.Name(rawValue: callNotificationName), object: call)
     }
 
     internal func reportMissedCall(_ call: SignalCall, callerName: String) {
+        AssertIsOnMainThread()
+
         notificationsAdapter.presentMissedCall(call, callerName: callerName)
     }
 
+    // TODO: who calls this?
```

Comment:
It's used by a UILocalNotification action. I'd say don't look into it much at this point as I'm making changes to it in the NSUserActivity feature I'm working on. I can double check that notification actions are working as part of that.